node/src/main.rs

```diff
@@ -866,70 +880,209 @@ async fn main() {
 }
 
 /// Parses an Ethereum connection string and returns the network name and Ethereum adapter.
-fn parse_ethereum_networks_and_nodes(
+fn parse_ethereum_networks(
     logger: Logger,
     networks: clap::Values,
     connection_type: ConnectionType,
     registry: Arc<MetricsRegistry>,
-) -> Result<HashMap<String, Arc<dyn EthereumAdapterTrait>>, Error> {
+) -> Result<EthereumNetworks, Error> {
     let eth_rpc_metrics = Arc::new(ProviderEthRpcMetrics::new(registry));
-    networks
-        .map(|network| {
-            if network.starts_with("wss://")
-                || network.starts_with("http://")
-                || network.starts_with("https://")
-            {
-                return Err(format_err!(
-                    "Is your Ethereum node string missing a network name? \
+    networks.fold(Ok(EthereumNetworks::new()), |networks, network_arg| {
```

Comment:
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.try_fold seems like the right thing to do here.

Reply:
Yeah, that cleans it up a bit.